### ISSUE TITLE
FR Add error code in messaging.ErrorInfo

### ISSUE THREAD

**Issue by p1ass** [firebase/firebase-admin-go#332]
### Describe your environment
  * Library version: v3.12.0
  * Firebase Product: Messaging 

### Describe the problem

I would like to switch errors handling according to the response of `SubscribeToTopic()`. 
However, `messaging.ErrorInfo` which is part of the response has only string reason, but not error code. So it is difficult to handle errors.
I think `messaging.ErrorInfo` should have an error code to make error handling more easier.

Since the error code has already used inside the library, it is not difficult to change it to be public, I think.
https://github.com/firebase/firebase-admin-go/blob/master/messaging/messaging.go#L58-L106  

#### Relevant Code:

The suggested change is below:

```go
type ErrorInfo struct {
	Index  int
	Reason string
    Code string // Added
}
```

https://github.com/firebase/firebase-admin-go/blob/master/messaging/messaging.go#L895-L899

**Comment by hiranya911** [comment]
Sounds like a reasonable addition to me.

**Comment by p1ass** [comment]
@hiranya911 Can I send a pull request about this issue?

**Comment by hiranya911** [comment]
@p1ass yes, that would be great!

**Comment by hiranya911** [comment]
This has been fixed for a while (see #343), and released in v4
